code/modules/mob/mob.dm

```diff
@@ -1347,3 +1349,37 @@
 		client.movingmob.client_mobs_in_contents -= src
 		UNSETEMPTY(client.movingmob.client_mobs_in_contents)
 		client.movingmob = null
+
+
+///Creates an image to show the mob's name on
+/mob/proc/create_name_image()
+
+	name_image = image(loc = src, layer = CHAT_LAYER)
+	name_image.plane = MOB_NAME_PLANE
+	name_image.appearance_flags = APPEARANCE_UI | KEEP_APART
+	name_image.pixel_y = -12
+	name_image.maptext_height = 40
+	name_image.maptext_width = 96
+	name_image.maptext_x = (96 - bound_width) * -0.5
+	update_name_image()
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_NAME_STEALTHY), .proc/hide_name)
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_NAME_STEALTHY), .proc/show_name)
+
+	add_overlay(name_image) //gonna cry?
```

Comment:
This looks like it's going to add it to the mob itself, and not just the client pressing it??

Reply:
@Mothblocks visibility is handled by planemaster which is on the client